compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -505,8 +509,15 @@ generateSrcFromLf env = noLoc mod
         -- convDataCons.
         [dataTypeCon0] <- [LF.unTypeConName dataTypeCon]
         let occName = mkOccName varName (T.unpack dataTypeCon0)
-        [ mkDataDecl env thisModule occName dataParams =<<
-            convDataCons dataTypeCon0 dataCons ]
+        pure $ do
+            ctxt <- noLoc <$> do
+                if NM.name dtype `NM.member` LF.moduleInterfaces (envMod env) then do
+                    interface <- mkGhcType env "DamlInterface"
+                    pure [noLoc interface]
+                else
+                    pure []
```

Comment:
This `DamlInterface` context is required so you can implement an interface defined in a data-dependency, right? Maybe worth adding a comment.

Reply:
exactly! will do